src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -2009,7 +2009,19 @@ public override bool Equals(object obj)
         }
 
         public override int GetHashCode()
-            => HashCode.Combine(MethodTargetType, ParameterTypes, GenericTypeParameters);
+        {
+            // algorithm based on https://stackoverflow.com/questions/263400/what-is-the-best-algorithm-for-an-overridden-system-object-gethashcode
+            unchecked
+            {
+                int result = 61;
+
+                result = result * 397 + (MethodTargetType != null ? MethodTargetType.GetHashCode() : 0);
+                result = result * 397 + ParameterTypes.SequenceGetHashCode();
+                result = result * 397 + GenericTypeParameters.SequenceGetHashCode();
+
+                return result;
+            }
```

Comment:
I would go with this since it seems to be the preferred API now. But I also don't know enough about the specifics of hashcodes to know why one is or isn't better (partially why I'd personally stick to the API that does it "properly" for you)

```suggestion
            HashCode hashCode = new();
            hashCode.Add(MethodTargetType);
            if (ParameterTypes is not null)
            {
                foreach (Type parameterType in ParameterTypes)
                {
                    hashCode.Add(parameterType);
                }
            }
            else
            {
                hashCode.Add(0);
            }
            
            if (GenericTypeParameters is null)
            {
                hashCode.Add(0);
                return hashCode.ToHashCode();
            }
            
            foreach (Type genericParameter in GenericTypeParameters)
            {
                hashCode.Add(genericParameter);
            }
            
            return hashCode.ToHashCode();
```

Reply:
> but Int32.GetHashCode() simply returns the integer itself

Also given this, I guess we now can replace the utility method `Utils.CombineHashCodes` in PowerShell with `HashCode.Combine`.

Reply:
> But I also don't know enough about the specifics of hashcodes to know why one is or isn't better (partially why I'd personally stick to the API that does it "properly" for you)

1. Less collisions.
2. Random seed per process (protects against cache attacks).